configurations/test_pipeline_configuration.py

```diff
@@ -84,18 +92,58 @@
             write_mode=WriteModes.CONCATENATE_TEXTS
         )
     ),
-    analysis_config=[
+    analysis_dataset_config=[
         AnalysisDatasetConfiguration(
             engagement_db_datasets = ["s01e01"],
-            dataset_type = DatasetTypes.RESEARCH_QUESTION_ANSWER
+            dataset_type = DatasetTypes.RESEARCH_QUESTION_ANSWER,
+            analysis_dataset = "s01e01",
+            coding_configs = [
+                CodingConfiguration(
+                    code_scheme = load_code_scheme("s01e01"),
+                    analysis_file_key = 's01e01'
```

Comment:
What's the difference between analysis_dataset and analysis_file_key? Can this be clearer with better variable names?

Reply:
analysis_dataset  is used to reference all messages for that analysis_config regardless of engagement_db_dataset they belong to. analysis_file_key will be used to export labels in the final analysis files e.g location -> constituency, county. Have renamed analysis_dataset  -> messages_analysis_dataset but welcome for suggestions.